engine/classes/Elgg/EntityIconService.php

```diff
@@ -435,6 +435,11 @@ public function getSizes($entity_type = null, $entity_subtype = null, $type = 'i
 			$sizes = $this->hooks->trigger("entity:$type:sizes", $entity_type, $params, $sizes);
 		}
 
+		if (empty($sizes)) {
```

Comment:
The empty array is caused by using a custom icon type, e.g:
```
$entity->saveIconFromUploadedFile('image', 'cover');
```
...but not defining the sizes for it with `entity:<type>:sizes, <entity_type>` hook. In this case the method returns the empty array defined on the line 422.

But yes, I guess the whole feature is useless if the hook is not being used, so throwing an exception might make more sense than logging an error.